lib/semantics/tools.h

```diff
@@ -57,6 +58,14 @@ const DeclTypeSpec *FindParentTypeSpec(const Symbol &);
 // Return the Symbol of the variable of a construct association, if it exists
 const Symbol *GetAssociationRoot(const Symbol &);
 
+enum class Tristate { No, Yes, Maybe };
+inline Tristate ToTristate(bool x) { return x ? Tristate::Yes : Tristate::No; }
```

Comment:
Should this be move to `common/idioms.h` or somewhere like that? There are a couple of other places we use `std::optional<bool>` where this might be better.

Reply:
How is `Tristate` better than using `optional<bool>`, which gets the job done with standard language features?

Reply:
With `std::optional<bool>` the three states aren't accessed symmetrically: you have to check for the first (`std::nullopt`) before you can check for the other two. In this case I have three equally valid states -- one shouldn't be distinguished.

I started with `std::optional<bool>` here but I thought it made the code less clear.